libp2p/Host.cpp

```diff
@@ -522,32 +522,50 @@ void Host::addNode(NodeID const& _node, NodeIPEndpoint const& _endpoint)
 
     if (_node == id())
     {
-        cnetdetails << "Ignoring the request to connect to self " << _node;
+        cnetdetails << "Ignoring request to connect to self " << _node;
         return;
     }
 
-    if (_endpoint.tcpPort() < 30300 || _endpoint.tcpPort() > 30305)
+    if (!isAllowedEndpoint(_endpoint))
```

Comment:
We shouldn't do this here, if we do, adding the node of the private network from LAN won't be possible without `--allow-local-discovery`

Actually `isAllowedEndpoint` might be better named something like `isEndpointAllowedForDiscovery`

Reply:
This was intentional - do we want to allow nodes on the private LAN because it's useful when running a private chain? Or is it because there's no benefit to preventing LAN nodes as peers?

Reply:
Yeah it's a common scenario for a private network - running several nodes in the local network, connect them using `--peerset` or RPC method

Reply:
Makes sense, thanks for the clarification!